Tentacle/Content.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  Content.swift
+//  Tentacle
+//
+//  Created by Romain Pouclet on 2016-11-28.
+//  Copyright © 2016 Matt Diephouse. All rights reserved.
+//
+
+import Foundation
+import Argo
+import Curry
+import Runes
+
+public enum Content {
+    public struct File: CustomStringConvertible {
+        public enum ContentType: String {
+            case file
+            case dir
+            case symlink
+            case submodule
+        }
+
+        /// The type of content
+        public let type: ContentType
+
+        /// Path to actual content when content is a symlink
+        public let target: String?
```

Comment:
It seems like we should move this into the innermost `enum`.

```swift
struct File {
    enum Content {
        case file(size: Int)
        case directory
        case symlink(target: String)
        case submodule(url: String)
    }

    let content: Content
}
```

What do you think?